tests/TestCollectors.py

```diff
@@ -12,33 +12,52 @@
 from vrops_exporter import run_prometheus_server
 from tools.YamlRead import YamlRead
 from VropsCollector import VropsCollector
+from tools.Resources import Resources
+from resources.Vcenter import Vcenter
+
 
 class TestCollectors(unittest.TestCase):
     def test_environment(self):
-        self.assertTrue(os.getenv('USER'),'no dummy USER set')
-        self.assertTrue(os.getenv('PASSWORD'),'no dummy PASSWORD set')
+        self.assertTrue(os.getenv('USER'), 'no dummy USER set')
+        self.assertTrue(os.getenv('PASSWORD'), 'no dummy PASSWORD set')
 
     def test_collector_metrics(self):
         metrics_yaml = YamlRead('tests/metrics.yaml').run()
         print(metrics_yaml)
-        #every collector got to be tested in here
-        random_prometheus_port = random.randrange(9000,9700,1)
+
+        # every collector got to be tested in here
+        random_prometheus_port = random.randrange(9000, 9700, 1)
         print("chosen testport: " + str(random_prometheus_port))
         for collector in metrics_yaml.keys():
+            print()  # nicer output
             print("\nTesting " + collector)
+
+            Vcenter.add_datacenter = MagicMock()
+            VropsCollector.create_resource_objects = MagicMock(return_value=Vcenter)
+            VropsCollector.get_adapter = MagicMock(return_value='{"name": "vcenter1", "uuid": "5628-9ba1-55e84701"}')
```

Comment:
Looks little bit confusing to me. Why would get_adapter give back vcenter1? Wouldn't that rather be Vcenter.uuid and Vcenter.name? 

What is an adapter here?

Reply:
Or should that be VropsCollector.get_vcenter rather than adapter?

Reply:
In vrops, a vcenter is an adapter type. I kept the vmware syntax to avoid confusing those who are familiar with vmware. 
For my understanding `get_vcenter` is also the better name. I will change it.

Reply:
ok got it, maybe put in a comment that this is originally referring to adapters in vmware

Reply:
`get_adapter` returns a list of dicts like _[{'name': 'vcenter', 'uuid': '24543647' }, ...]_

the `create_resource_objects` makes from the get_requests objects. So you can call these like vcenter.name, vcenter.uuid

Reply:
I would prefer Richards first initial approach with `get_adapter`, 'cause this is technically correct. I mean, in the end the whole purpose of our construct is, that no one has to take care of the exporter itself, if he wants to contribute a module(/metric). In this case no one has to dig into the `get_adapter` method.
vRops is capable of monitoring more than just vCenters. If this is needed at some point, we would add more **Adapters**. `get_adapter` would better fit in that case.
Just add as comment "In VMware environments a vCenter is an adapter." (Similar to what Tommy has written).

Reply:
Convinced, back to `get_adapter`

Reply:
ok, I roll back my changes

Reply:
resolved